---
title: "EPID 674 - Lecture 9 - Assignment"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE, eval = FALSE)
```

__Load the Tidyverse__

* We have to explicitly load the packages that we will use for our analysis.

```{r load_tidyverse, echo=TRUE, eval=TRUE}

library(tidyverse)

```

***

__Install the DirichletMultinomial Package via Bioconductor__

* Bioconductor is a repository of bioinformatics software.

```{r load_dirichletmultinomial, echo=TRUE, eval=TRUE}

#source("https://bioconductor.org/biocLite.R")
#biocLite("DirichletMultinomial")
library(DirichletMultinomial)

```

***

__Install the Viridis Package for Heatmaps__

* Viridis provides color palettes that (1) span a wide color range, making differences easy to see; (2) are perceptually uniform, meaning that values close to each other have similar-appearing colors and values far away from each other have more different-appearing colors; and are (3) robust to colorblindness, so that the above properties hold true for people with common forms ofcolorblindness, as well as in grey scale printing.

```{r load_viridis, echo=TRUE, eval=TRUE}

#install.packages("viridis")
library(viridis)

```

***

__Load & Tidy the Data__

* We will again evaluate an OTU table from a study of respiratory tract microbiome (OP = oropharyngeal, ET = endotracheal), which is described here: https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-016-0151-8

```{r load_otus, echo=TRUE, eval=TRUE, fig.align='center'}

icu_otus <- read_tsv(file = "./data/icu_de_novo_otu_table_op_et.txt") # we skip the first line because it holds a file descriptor, not tabular data

icu_assignments <- read_tsv(file = "./data/icu_de_novo_assignment_ltp_genus.txt")

icu_otus %>%
  dplyr::rename(otu_id = '#OTU ID') %>%
  gather(key = "specimen_id", value = "read_count", -otu_id) %>%
  left_join(icu_assignments, by = "otu_id") %>%
  mutate(subject_id = str_extract(string = specimen_id, pattern = "001|002|003|004|005|006|007|008|009|010|011|012|013|014|015"),
         specimen_type = str_extract(string = specimen_id, pattern = "ET|OP"),
         collection_date = as.Date(str_sub(string = specimen_id, start = 12, end = 19), format = "%Y%m%d")) %>%
  group_by(subject_id) %>%
  mutate(days_since_enrollment = as.numeric(collection_date - min(collection_date))) %>%
  ungroup() -> tidy_icu_final

tidy_icu_final %>%
  dplyr::slice(1:11)

```

***

__Questions__

1. Read and tidy the ICU OTU data, select only the "OP" specimens, form a matrix from the tidy data, and select only those OTUs that have > 500 total reads:



2. Evaluate a Dirichlet-Multinomial Mixture (DMM) model for the "OP" specimen matrix and assign each specimen a community/mixture type.


3. Which OTUs best distinguish the DMM types? Using your original tidy data, what taxonomic assignments do these OTUs have?


